### PR TITLE
Add mutation from `@foo` to `foo`

### PR DESCRIPTION
**Pull request by backus**
In the first commit I've reconfigured the AUOM integration to not expect full mutation coverage since this commit adds mutations which AUOM does not cover

### CODE REVIEW

**Comment by backus** [comment]
Here are the mutations in AUOM revealed by this PR: 

```diff
@@ -1,15 +1,15 @@
 def pretty_unit
   if unitless?
     return
   end
-  numerator = Inspection.prettify_unit_part(@numerators)
+  numerator = Inspection.prettify_unit_part(numerators)
   denominator = Inspection.prettify_unit_part(@denominators)
   if numerator.empty?
     numerator = "1"
   end
   if denominator.empty?
     return " #{numerator}"
   end
   format(UNIT_FORMAT, numerator, denominator)
 end
```

```diff
@@ -1,15 +1,15 @@
 def pretty_unit
   if unitless?
     return
   end
   numerator = Inspection.prettify_unit_part(@numerators)
-  denominator = Inspection.prettify_unit_part(@denominators)
+  denominator = Inspection.prettify_unit_part(denominators)
   if numerator.empty?
     numerator = "1"
   end
   if denominator.empty?
     return " #{numerator}"
   end
   format(UNIT_FORMAT, numerator, denominator)
 end
```

```diff
@@ -1,13 +1,13 @@
 def initialize(scalar, numerators, denominators)
   @scalar = scalar
   [numerators, denominators].permutation do |left, right|
     left.delete_if do |item|
       right.delete_at(right.index(item) || right.length)
     end
   end
   @numerators = numerators.freeze
   @denominators = denominators.freeze
-  @unit = [@numerators, @denominators].freeze
+  @unit = [numerators(), @denominators].freeze
   @scalar.freeze
 end
```

```diff
evil:AUOM::Unit#initialize:/Users/johnbackus/Dropbox/coding/auom/lib/auom/unit.rb:248:f1c53
@@ -1,13 +1,13 @@
 def initialize(scalar, numerators, denominators)
   @scalar = scalar
   [numerators, denominators].permutation do |left, right|
     left.delete_if do |item|
       right.delete_at(right.index(item) || right.length)
     end
   end
   @numerators = numerators.freeze
   @denominators = denominators.freeze
-  @unit = [@numerators, @denominators].freeze
+  @unit = [@numerators, denominators()].freeze
   @scalar.freeze
 end
```

```diff
evil:AUOM::Unit#initialize:/Users/johnbackus/Dropbox/coding/auom/lib/auom/unit.rb:248:e489b
@@ -1,13 +1,13 @@
 def initialize(scalar, numerators, denominators)
   @scalar = scalar
   [numerators, denominators].permutation do |left, right|
     left.delete_if do |item|
       right.delete_at(right.index(item) || right.length)
     end
   end
   @numerators = numerators.freeze
   @denominators = denominators.freeze
   @unit = [@numerators, @denominators].freeze
-  @scalar.freeze
+  scalar().freeze
 end
```

**Comment by dkubb** [comment]
@mbj I'm curious if it would be possible to emit these methods without the empty parens?

**Comment by backus** [comment]
@mbj this is ready for review

**Comment by mbj** [comment]
> @mbj I'm curious if it would be possible to emit these methods without the empty parens?

No. Because unparser is lvar scope aware and needs to use the empty parents here to disambiguate the private getter with the lvar.

By default argument less getter calls are emitted without parens in unparser.

**Comment by mbj** [comment]
> Here are the mutations in AUOM revealed by this PR:

@backus The build will not pass unless `auom` is fully mutation covered. We need to fix auom first. I'll do this before merging that one.

**Comment by mbj** [comment]
Interesting it *still* passed, it should not.

**Comment by backus** [comment]
> Interesting it still passed, it should not. 

Note https://github.com/backus/mutant/commit/05de96219351b892ac8501675a6996b69b3f8abd

**Comment by mbj** [comment]
> Interesting it still passed, it should not.
> Note backus@05de962

Hah, was not looking at the diff yet. Did not expected you'd change this. Lets fix auom before introducing a "regression" to the corpus test.

**Comment by backus** [comment]
> Lets fix auom before introducing a "regression" to the corpus test.

:thumbsup: I'll drop the commit in that case

**Comment by backus** [comment]
Dropped the config change

File: lib/mutant/mutator/node/named_value/access.rb [hunk 1]
```diff
@@ -19,6 +19,44 @@ def dispatch
...
+            end
+
+            private
+
+            # Emit instance variable as attribute send
+            #
+            # @return [undefined]
+            #
+            # @api private
+            def emit_attribute
```
Comment (mbj) [inline lib/mutant/mutator/node/named_value/access.rb]: The name of the method is very generic. Lets name it something like `emit_attribute_read`?

File: lib/mutant/mutator/node/named_value/access.rb [hunk 2]
```diff
@@ -19,6 +19,44 @@ def dispatch
...
+              emit(s(:send, nil, attribute_name))
+            end
+
+            # Variable name without leading '@'
+            #
+            # @return [Symbol]
+            #
+            # @api private
+            def attribute_name
+              name.slice(1..-1).to_sym
```
Comment (mbj) [inline lib/mutant/mutator/node/named_value/access.rb]: I think the range literal should be moved to a constant.

We typically use constants for all mutable literals, unless the propose of the literal is to create a mutable value.

Comment (mbj) [inline lib/mutant/mutator/node/named_value/access.rb]: Ranges are monkeypatchabe, thus mutable :(

**Comment by mbj** [comment]
@backus I did a small code only pass, fix for auom is in progress.

File: lib/mutant/mutator/node/named_value/access.rb (hunk 2 above)
Comment (dkubb) [inline lib/mutant/mutator/node/named_value/access.rb]: Another nice side effect of moving literals to constants is you become more aware of their usage in code. Over-usage of string literals or other promitives in code is often a sign of other code smells being present, such as [Primitive Obsession] (http://c2.com/cgi/wiki?PrimitiveObsession) or [Stringly Typed](http://c2.com/cgi/wiki?StringlyTyped) coding.

Not saying that's the case here, just something to be aware of.

**Comment by backus** [comment]
@mbj fixed flags.  Should be ready for another pass

**Comment by mbj** [comment]
Closing in favor of #463 as I've direct control over this branch. @backus Your commit authorship is keept intact.

**Comment by mbj** [comment]
@backus I just realize I could have pushed to your fork as you gave me access, sorry for the noise.

**Pull request closed without merging**